algebird-core/src/main/scala/com/twitter/algebird/CorrelationGroup.scala

```diff
@@ -0,0 +1,107 @@
+package com.twitter.algebird
+
+object Correlation {
+  def apply(x: (Double, Double)): Correlation =
+    Correlation(c2 = 0, m2Left = 0, m2Right = 0, m1Left = x._1, m1Right = x._2, 1L)
+
+  implicit val group: Group[Correlation] = CorrelationGroup
+}
+
+/**
+ * A class to calculate covariance and the first two central moments of a sequence of pairs of Doubles, from which the
+ * pearson correlation coeifficient can be calculated.
+ *
+ * m{i}Left denotes the ith central moment of the first projection of the pair.
+ * m{i}Right denotes the ith central moment of the second projection of the pair.
+ * c2 the covariance equivalent of the second central moment, i.e. c2 = Sum_(l,r) (l - m1Left)*(r - m1Right).
+ *
+ */
+case class Correlation(
+    c2: Double,
+    m2Left: Double,
+    m2Right: Double,
+    m1Left: Double,
+    m1Right: Double,
+    m0: Long
+) {
+  def count: Long = m0
+
+  def meanLeft: Double = m1Left
+
+  def meanRight: Double = m1Right
+
+  // useful in calculating variance and stddev; private because we don't have skewness or kurtosis info
```

Comment:
I'd rather move this code to the `Moment` object so we don't have to allocate a `Moments` class to get the result. The JIT *may* be able to allocate on the stack using escape analysis, but I'm nervous.... but if you don't have time, I understand (that said, my request to use a Double for the m0 may mean we can't use this method).

Reply:
@johnynek  I agree re:weights, and it's not very deep business logic, so I that re-implementing it here is no big deal. However, what should be the equivalent of this line? Should `> 1` still be the threshold in the presence of weights? 

https://github.com/twitter/algebird/blob/cf913f69992bf4345c3d38a46698c3c7278c6d3f/algebird-core/src/main/scala/com/twitter/algebird/MomentsGroup.scala#L35

Reply:
no, I think it should probably be `count != 0.0` I suppose. But even then, maybe we should just remove the branch and always return the value and let the caller be careful what it means if count <= 0.0?

Reply:
I opted for the latter, but am open to changing it again.